### ISSUE TITLE
A separate version of React without support for classes

### ISSUE THREAD

**Issue by ecreeth** [reactjs/rfcs#116]
The objective with this is that there are two versions of React.

- With support for both functional components and class components, as it currently remains.
- With unique support for functional components.

This will allow developers to continue working on their legacy projects by implementing the full version of React.

While in the new projects they may choose to use only class components or functional components.

This last would bring a considerable reduction of the bundle size.

**Comment by marvinhagemeister** [comment]
_Disclaimer: I work on [preactjs](https://github.com/developit/preact/) an 3kB alternative where we treat size as our top priority_

Since hooks have been announced we got several requests if we could ship a hooks-only version alongside the current one. Dropping the code for classes and replacing it with hooks leads to some savings, but not as much as people tend to think it does.

Despite our focus on size, we backed out of that idea for now and postponed it for a later time. The main issue is the burden of maintenance overhead of supporting 2 versions of a framework. Both classes and hooks are somewhat tightly integrated into the reconciler and making it pluggable would likely defeat potential size savings.

Then there is also the issue of a less ideal developer experience. The current ecosystem is huge, and there is a non-trivial amount of components on npm that is written with classes and won't receive an update. Same for facebook which is made of 15.000+ components, which would need to be updated. For new users we (at preactjs) fear that having to be aware of which npm packages will work or not is too much for new users. We'd rather have everything work out of the box.

Again, I don't work with react team and they may have a different opinion on that matter :slightly_smiling_face:
